src/main/java/frc/robot/subsystems/Mechanism.java

```diff
@@ -0,0 +1,374 @@
+        this.m_topBeltMotor.setSmartCurrentLimit(40, 20); //keep 40 for neo550
+        this.m_sourceMotor.setSmartCurrentLimit(60, 20);
+        this.m_ampMotor.setSmartCurrentLimit(60, 20);
+
+        this.m_bottomBeltMotor.setInverted(true);
+        this.m_topBeltMotor.setInverted(true);
+        this.m_sourceMotor.setInverted(false);
+        this.m_ampMotor.setInverted(false);
+
+        this.m_bottomBeltMotor.setOpenLoopRampRate(0.05);
+        this.m_topBeltMotor.setOpenLoopRampRate(0.05);
+        this.m_sourceMotor.setOpenLoopRampRate(0.05);
+        this.m_ampMotor.setOpenLoopRampRate(0.05);
+
+        this.m_bottomBeltMotor.enableVoltageCompensation(12);
+        this.m_topBeltMotor.enableVoltageCompensation(12);
+        this.m_sourceMotor.enableVoltageCompensation(12);
+        this.m_ampMotor.enableVoltageCompensation(12);
+
+        requestIntakeType = 0;
+    }
+
+    /**
+     * Gets the instance of the mechanism
+     * @return the instance of the mechanism
+     */
+    public static Mechanism getInstance() {
+        if (instance == null) {
+            instance = new Mechanism();
+        }
+        return instance;
+    }
+
+    /**
+     * updates the phase of the beam break sensor
+     */
+    private void updatePhase() {
+        this.m_beamBreak.updatePhase();
+        this.m_currentPhase = this.m_beamBreak.getPhase();
+    }
+
+    /**
+     * Sets the speed of the belt motor
+     * @param speed the speed in volts [0 --> 12]
+     */
+    private void setBeltSpeed(double speed) {
+        this.m_bottomBeltMotor.setVoltage(speed);
+        this.m_topBeltMotor.setVoltage(speed);
+    }
+
+    /**
+     * Sets the speed of the source motor
+     * @param speed the speed in volts [0 --> 12]
+     */
+    private void setSourceSpeed(double speed) {
+        this.m_sourceMotor.setVoltage(speed);
+    }
+    
+    /**
+     * Sets the speed of the amp motor
+     * @param speed the speed in volts [0 --> 12]
+     */
+    private void setAmpSpeed(double speed) {
+        this.m_ampMotor.setVoltage(speed);
+    }
+
+    /**
+     * A method to check if the phase changed
+     * @param phase the phase that is checked
+     */
+    public boolean checkState(Phase phase) {
+        return m_currentPhase == phase;
+    }
+
+    /**
+     * A function for requesting ground/source intake using LEDs
+     * @param intakeType The requested intake type (1 is ground, 2 is source, 0 is nothing)
+     */
+    public void requestIntake(int intakeType) {
+        // Set the requested intake based on the input
+        if (requestIntakeType == intakeType) { // If you request the same intake twice the lights turn off
+            requestIntakeType = 0;
+        }
+        else {
+            requestIntakeType = intakeType;
+        }
+
+        // Set the LED color to the requested intake
+        if (this.m_currentPhase != Phase.NONE) {
+            requestIntakeType = 0;
```

Comment:
Consider adding `return;` to next line, then make following else ifs into a switch case.